### PR TITLE
Fixed trying to access non-existing properties in strict_variables mode

### PR DESCRIPTION
**Pull request by supersmile2009**
I am targeting this branch, because this is a backwards-compatible bug fix

## Changelog

```markdown
### Fixed
- Error on some fields in collection table when `strict_variables` mode is enabled
```

## Subject
When `strict_variables` mode is enabled, trying to access non-existing array key from a template results in exception.
```
                        {% if nested_field.vars['required'] %}
                            class="required"
                        {% endif %}
```

`nested_field.vars['required']` and `nested_field.vars.errors` can be missing if collection form contains some a not mapped field with field type defined.

## Example
FooAdmin
```
    protected function configureFormFields(FormMapper $formMapper): void
    {
        $formMapper
            ->add(
                'bars',
                CollectionType::class,
                ['by_reference' => false],
                ['edit' => 'inline', 'inline' => 'table']
            )
            ->end();
    }
```
BarAdmin
```
    protected function configureFormFields(FormMapper $formMapper): void
    {
        $formMapper
            ->add('assetName')
            ->add(
                'button',
                ButtonType::class,
                [
                    'href' => $asset->getUrl(),
                    'hideOnDisabled' => true,
                    'enabled' => (bool) $asset->getUrl(),
                    'label' => "Play",
                ]
            )
            ->end();
    }
```
In this case `button` FormView will not have those properties because in `Symfony\Component\Form\FormBuilder::create()`
```
    public function create($name, $type = null, array $options = array())
    {
        // [... not relevant code]

        // This is going to be true
        if (null !== $type) {
            // This method doesn't run 'required' guesser and doesn't add 'required' option
            return $this->getFormFactory()->createNamedBuilder($name, $type, null, $options);
        }

        return $this->getFormFactory()->createBuilderForProperty($this->getDataClass(), $name, null, $options);
    }
```

## Solution
Handle check of potentially missing property using `is defined` Twig construct, same as it's done for `nested_field.vars['attr']['hidden']`

### CODE REVIEW

File: src/Resources/views/CRUD/Association/edit_many_to_many.html.twig [hunk 1]
```diff
@@ -21,7 +21,11 @@ file that was distributed with this source code.
                                         {% if field_name == '_delete' %}
                                             <th>{{ 'action_delete'|trans({}, 'SonataAdminBundle') }}</th>
                                         {% else %}
-                                            <th {{ nested_field.vars['required']  ? 'class="required"' : '' }}>
+                                            <th
+                                                {% if (nested_field.vars['required'] is defined) and nested_field.vars['required'] %}
```
Comment (greg0ire) [inline src/Resources/views/CRUD/Association/edit_many_to_many.html.twig]: How about using the `default` filter instead?

Comment (supersmile2009) [reply src/Resources/views/CRUD/Association/edit_many_to_many.html.twig]: Done

**Comment by greg0ire** [comment]
Thanks @supersmile2009 ! Just to confirm, is this about new versions Twig generating php files that are in strict mode?

**Comment by greg0ire** [comment]
Thanks for the explanation

**Pull request merged**